mixins/loading-complete/loading-complete-mixin.js

```diff
@@ -1,21 +1,36 @@
-export const LoadingCompleteMixin = (superclass) => class extends superclass {
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
 
-	constructor() {
-		super();
-		this._loadingCompletePromise = new Promise(resolve => {
-			this._loadingCompleteResolve = resolve;
-		});
+export const LoadingCompleteMixin = dedupeMixin((superclass) => class extends superclass {
+
+	#loadingCompleteResolve;
+
+	// eslint-disable-next-line sort-class-members/sort-class-members
+	#loadingCompletePromise = this.getLoadingComplete === this.#getLoadingComplete
+		? new Promise(resolve => this.#loadingCompleteResolve = resolve)
+		: Promise.resolve();
+
+	get getLoadingComplete() {
+		return this.#getLoadingComplete;
+	}
+
+	get loadingComplete() {
+		return this.#getLoadingComplete();
+	}
+
+	get resolveLoadingComplete() {
+		return this.#resolveLoadingComplete.bind(this);
 	}
 
-	getLoadingComplete() {
-		return this._loadingCompletePromise;
+	async #getLoadingComplete() {
+		await super.getLoadingComplete?.();
+		return this.#loadingCompletePromise;
 	}
```

Comment:
Other mixins may already be extending this one, or may just have their own `getLoadingComplete` methods already, so we need to make sure those are called.

Reply:
I don't think it makes a difference since these are just the promises. The actual work has already been kicked off.